### ISSUE TITLE
Table of contents navigation on ingest form

### ISSUE THREAD

**Issue by sseymore** [OregonDigital/OD2#1202]
### Descriptive summary

The ingest form is long and a table of contents-like navigation would be useful for jumping to groups/sections for metadata input.

### Expected behavior

Have a floating table of contents in the right sidebar area underneath Save Work/Visibility-similar to:

![Screenshot from 2020-06-18 15-18-13](https://user-images.githubusercontent.com/7328774/85077708-c1de0900-b177-11ea-9681-94909758d34e.png)

https://css-tricks.com/sticky-table-of-contents-with-scrolling-active-states/
https://afeld.github.io/bootstrap-toc/
https://jsfiddle.net/gableroux/S2SMK/

### Related work

Add groupings to ingest form. See https://github.com/OregonDigital/OD2/issues/1201

### Accessibility Concerns

Not sure.

**Comment by straleyb** [comment]
QA:

- [ ] Typing "https://staging.oregondigital.org/concern/generics/new?locale=en#formats" should take you down to formats (materials)

**Comment by jsimic** [comment]
@sseymore @wickr @carakey @KevinJonesMeta 
QA pass. HOWEVER, due to my tiny laptop screen the ToC requires me to scroll to the very bottom of the form in order to see it, rendering it useless for navigation. Can we move it above the Save Work/Visibility box? Can we compact the box? Will discuss at the next Metadeities meeting.

**Comment by straleyb** [comment]
QA: 
- [ ] Types exists and links
- [ ] required fields exists and links

Accessibility can be refined. We can have @azle look at it and see if there is any design choices that would work best in this area.

**Comment by azle** [comment]
**Accessibility Concerns and Look and Feel**

For the TOC, I recommend the following.

- Place the TOC in a similar box as the look and feel for "Requirements and Visibility" content. White background with a light gray border. With the links listed. 
- I would also include a header for this section. 
- The link colors should be the same look and feel as active links on the site. Latos 16 Heavy #073c5c

**Comment by straleyb** [comment]
QA: The ToC should work. As you scroll down the works page, it should highlight the section you are in. You should also be able to click a section in the ToC and it will take you to that place on the page.

**Comment by straleyb** [comment]
@azle Might want to check it out once it is live as well to see if it meets accessibility criteria.

**Comment by sseymore** [comment]
@straleyb Everything looks good, but I don't see the highlight on the sections as you scroll?

**Comment by straleyb** [comment]
@sseymore It should be good now. Just hadn't gotten merged yet.

**Comment by sseymore** [comment]
@straleyb I am seeing the highlight as I navigate the ToC but no shading. I am using Chrome. Also, the link to "Geographics" is working now, but the label is still wrong as referenced above. 

I met with @azle to discuss accessibility concerns with placement of ToC and no header for it and need to discuss these changes with the Metadata team tomorrow before we make any more changes. Will update @azle tomorrow after that meeting and leave comments here.

**Comment by azle** [comment]
Issues that I see but not sure priority for fix.

1. Tabs are not going to appropriate page via keyboard. Works currently by clicking and using the mouse. (Descriptions, Files, Relationships)
2. A keyboard user would have to tab through entire form (middle column) before getting to third column.
3. Having the box for the table content below the browser fold in smaller screens a user would not even realize its there. It also becomes unusable for someone using a keyboard to navigate.
4. Jump to box needs a header like the other sections in the third column, maybe Table of Contents

**Comment by sseymore** [comment]
@azle @straleyb The metadata team is OK with higher contrast in the ToC and a header for it. @wickr had some concerns about placing it above the Save Work area.

**Comment by wickr** [comment]
For #2, I'm not sure what the expectation for this should be. I see the TOC as a navigation helper, so I would think, say, keyboard navigation would stay on the main form, or go there first. In other layouts/systems, this TOC would be 'first' on the page, but I don't know that we need that here.

That ties in with #3, I don't think it's great to move this above the 'Save Work' box. That has more critical elements in it like privacy settings and form checks. Since the TOC isn't an essential navigation element, it being below the fold is ok. (and since the form is so tall, it's likely to be discovered quickly)
